rest/server.go

```diff
@@ -119,6 +119,7 @@ type HTTPServer struct {
 	services    map[string]Service
 	evtHandlers map[ServerEvent][]ServerEventFunc
 	lock        sync.RWMutex
+	gracefulShutdownTimeoutSeconds uint32
```

Comment:
NIT:maybe just `shutdownTimeout` ?

Reply:
I like the graceful naming to shutdown, that is what the stop server is trying to do and i am being explicit about it. I will remove seconds from name and replace type uint32 with duration.

Reply:
NIT: i agree, but for example `httpServer.Shutdown` method is not called `httpServer.GracefulShutdown` even though it does graceful shutdown (https://golang.org/pkg/net/http/#Server.Shutdown).

I guess what I am trying to say is that long names are not really GoLang style. E.g. in this particular case naming it `shutdownTimeout` and the method `WithShutdownTimeout` would make more sense since it is kind of obvious that it is about `graceful` shutdown. But i am all for the name you have chosen too :)